### ISSUE TITLE
treatments for Idiopathic Juvenile Osteoporosis do not come back when searching for Bone Disease

### ISSUE THREAD

**Issue by sierra-moxon** [NCATSTranslator/Feedback#52]
In MONDO, `idiopathic juvenile osteoporosis` is a subclass_of `bone disorder` :
<img width="737" alt="Screen Shot 2022-12-09 at 8 19 51 AM" src="https://user-images.githubusercontent.com/4561292/206746115-f2b5346d-f07f-4896-912d-4abf5cd763fc.png">

I would expect that treatments for  `idiopathic juvenile osteoporosis` should be returned when searching for treatments of `bone disease` (with the caveat that the returned disease should match the disease treated by the result, not the parent class!  Because of this, I would expect that the disease listed in the UI would accurately reflect the disease to which the treatment was applied.  

In the example above, if I searched for `bone disease` I would expect to see back a drug that when I expanded the result, would show paths from the drug to `idiopathic juvenile osteoporosis` (a nice extra would be an indication in the results that `idiopathic juvenile osteoporosis` is a subclass of `bone disease` -- my search term). 

Right now we do not return treatments for subclasses of the disease queried, for example:
search for `idiopathic juvenile osteoporosis`, one of many results is `Dihydrotachysterol 3`.  Then search for `bone disease`, wait for the entirety of the results to return, reload those extra results, and then use the filter bar at the left of the results to search for `Dihydrotachysterol 3`. 

<img width="826" alt="Screen Shot 2022-12-09 at 8 28 59 AM" src="https://user-images.githubusercontent.com/4561292/206748796-80b861a5-b30c-4792-81ab-739386e87837.png">

<img width="792" alt="Screen Shot 2022-12-09 at 8 28 48 AM" src="https://user-images.githubusercontent.com/4561292/206748807-10ebdd09-f84c-427c-b1f4-712ba711b220.png">

**Comment by sierra-moxon** [comment]
From TAQA: we see the children treatments when searching for Osteoporosis (we see menopausal osteoporosis treatments). 
It could be that we aren't resolving enough answers, maybe its also ranked lower.

**Comment by sierra-moxon** [comment]
Sierra: put in the PKs for this result.

**Comment by sierra-moxon** [comment]
search for `bone disease`:
http://transltr-bma-ui-dev.ncats.io/results?q=8fd28442-7f44-49c0-a353-4cb700caa26b
ARAX id: 8fd28442-7f44-49c0-a353-4cb700caa26b

search for `Idiopathic Juvenile Osteoporosis`
http://transltr-bma-ui-dev.ncats.io/results?q=1c20a12f-f299-48e9-ae1f-5f4393d6d884
ARAX id: 1c20a12f-f299-48e9-ae1f-5f4393d6d884

**Comment by sierra-moxon** [comment]
search for `Idiopathic Juvenile Osteoporosis` 
http://transltr-bma-ui-dev.ncats.io/results?q=1c20a12f-f299-48e9-ae1f-5f4393d6d884

search for `bone disease`:
http://transltr-bma-ui-dev.ncats.io/results?q=5e9d00a6-8c4e-407c-9653-e5833fb20978

**Comment by sierra-moxon** [comment]
reviewing results via the ARAX interface, I only see `Dihydrotachysterol 3` being returned from Aragorn, ranked 355th via Aragorn.
<img width="362" alt="Screen Shot 2022-12-14 at 2 52 35 PM" src="https://user-images.githubusercontent.com/4561292/207732544-b2853d74-a53e-4bf7-8849-91622bedaa26.png">
the results make sense to me (others should review them)

I do not get a result from Aragorn for `bone disease` - could this be the issue @cbizon?  (it just spins, I let it run for 4+ hours this morning). 

tangentially, this edge from TMKP seems to be about premenopausal ` idiopathic osteoporosis.`  I am not sure if there is a difference between `juvenile` and `premenopausal` -- need to ask experts.
<img width="356" alt="Screen Shot 2022-12-14 at 2 54 44 PM" src="https://user-images.githubusercontent.com/4561292/207732859-95f22a80-2d86-4031-a846-ea5a0830693f.png">

**Comment by cbizon** [comment]
@sierra-moxon it looks like ARAGORN returned for 5e9d00a6-8c4e-407c-9653-e5833fb20978 and for 8fd28442-7f44-49c0-a353-4cb700caa26b.   Were those the ones you were waiting on?  (If so that does seem pretty slow, but it did finish eventually)

**Comment by sierra-moxon** [comment]
oh nice! :) I see the results there this morning, thanks!

**Comment by sierra-moxon** [comment]
From TAQA:
We need to have a plan for ARS/UI interaction when you return large numbers of answers.  
Might be an architecture issue, but low priority - hopefully someone would search for JO instead of Bone disorder.

**Comment by sierra-moxon** [comment]
Searching for osteoporosis does return treatments for its child class idiopathic juvenile osteoporosis.  Still does not return results when I filter the "bone disorder" results by "idiopathic juvenile osteoporosis".   This seems like a systemic issue to handle in the Fall.

**Comment by sstemann** [comment]
many of the treatments from the first page of [Idiopathic Juvenile Osteoporosis](https://ui.test.transltr.io/main/results?l=Idiopathic%20Juvenile%20Osteoporosis&i=MONDO:0019409&t=0&r=0&q=6b6acd51-8782-4b46-b61e-a82ccce97ef0) are in the results for [Bone Disorder](https://ui.test.transltr.io/main/results?l=Bone%20Disorder&i=MONDO:0005381&t=0&r=0&q=fee06168-91bc-462c-99b6-220de9d0a5ae)
